trade-finance-manager-ui/scripts/enable-find-utilisation-reports-by-year-dropdown-autocomplete.ts

```diff
@@ -0,0 +1,29 @@
+const enableFindUtilisationReportsByYearDropdownAutocomplete = (): void => {
+  const findUtilisationReportsByYearForm: HTMLFormElement | null = document.querySelector('form#find-utilisation-reports-by-year--form');
```

Comment:
Just had a look on other public scripts they seems to be referring to a DOM element using `data-cy` can the existing pattern please be used?

Reply:
Using `data-cy` in production is an antipattern as `data-cy` should only be used for testing

Reply:
I also can't find any TFM-UI scripts which use the `data-cy` tag to access any elements

Reply:
Just weighing in here as I asking for this to be an ID instead of `data-cy` initially. Cypress explicitly wants `data-cy` to only be for tests and not be referenced in either css or JS as per https://docs.cypress.io/guides/references/best-practices. I realise we've got an existing pattern of using it in portal at points but is keeping that pattern our preference over the cypress advice of how to use it?

Reply:
Further to this, [jquery best practices](https://www.freecodecamp.org/news/jquery-selectors) suggest using IDs as the primary way of selecting elements due to their high specificity (in contrast to other attributes such as `class` or `data-cy`)

>ID selectors have a high specificity making them difficult to override. Classes have a much lower specificity and are generally the preferred way to style elements in order to avoid specificity issues.

Reply:
Going to leave this as is and update other form selection under [FN-3390](https://ukef-dtfs.atlassian.net/browse/FN-3390)